File: codes/bt33.c
//Code by Mudit
//Date: 10/09/2026
#include <stdio.h>
#include <math.h>

int main()
{
    double x = 1.0;          // Given initial guess x0 = 1
    double next_x;
    int iterations = 6;

    printf("Newton-Raphson Method\n");
    printf("Equation: e^x - 2 = 0\n");
    printf("Difference equation: x[n+1] = x[n] - 1 + 2e^(-x[n])\n\n");

    printf("n\t x[n]\n");

    // Print the initial value
    printf("0\t %.8lf\n", x);

    // Apply the Newton-Raphson update repeatedly
    for (int n = 0; n < iterations; n++)
    {
        // x[n+1] = x[n] - 1 + 2e^(-x[n])
        next_x = x - 1.0 + 2.0 * exp(-x);

        printf("%d\t %.8lf\n", n + 1, next_x);

        // Use the new value in the next iteration
        x = next_x;
    }

    return 0;
}

File: codes/bt33.py
#Code by Mudit
#Date: 10/09/2026
import numpy as np
import matplotlib.pyplot as plt

# Original equation:
# f(x) = e^x - 2

x = np.linspace(-1, 2, 500)
y = np.exp(x) - 2

# Newton-Raphson difference equation:
# x[n+1] = x[n] - 1 + 2e^(-x[n])

N = 6

xn = np.zeros(N + 1)

# Initial guess
xn[0] = 1.0

for n in range(N):
    xn[n + 1] = xn[n] - 1 + 2 * np.exp(-xn[n])

n = np.arange(N + 1)


print("n\t x[n]")

for i in range(N + 1):
    print(i, "\t", xn[i])

# Plot both

plt.plot(x, y, label=r"$f(x)=e^x-2$")

plt.stem(n, xn, label="")

plt.axhline(0)

plt.xlabel("x / iteration number n")
plt.ylabel("f(x) / x[n]")

plt.title("Original Equation and Difference Equation")

plt.grid()
plt.legend()

plt.savefig("/sdcard/github/ID1063/codes/graph.pdf")

plt.close()

#opening the pdf
import subprocess
subprocess.run(["termux-open", "/sdcard/github/ID1063/codes/graph.pdf"])
